### ISSUE TITLE
BUG: Unable to import scipy signals (No module named 'scipy.optimize._root')

### ISSUE THREAD

**Issue by sophiehourihane** [scipy/scipy#19464]
### Describe your issue.

I am unable to import scipy signals. I get a module not found error. 

### Reproducing Code Example

```python
from scipy import signal
```

### Error message

```shell
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/sophie.hourihane/.conda/envs/bw_cpp_env_103123/lib/python3.10/site-packages/scipy/__init__.py", line 189, in __getattr__
    return _importlib.import_module(f'scipy.{name}')
  File "/home/sophie.hourihane/.conda/envs/bw_cpp_env_103123/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/home/sophie.hourihane/.conda/envs/bw_cpp_env_103123/lib/python3.10/site-packages/scipy/signal/__init__.py", line 324, in <module>
    from ._bsplines import *
  File "/home/sophie.hourihane/.conda/envs/bw_cpp_env_103123/lib/python3.10/site-packages/scipy/signal/_bsplines.py", line 12, in <module>
    from scipy.interpolate import BSpline
  File "/home/sophie.hourihane/.conda/envs/bw_cpp_env_103123/lib/python3.10/site-packages/scipy/interpolate/__init__.py", line 167, in <module>
    from ._interpolate import *
  File "/home/sophie.hourihane/.conda/envs/bw_cpp_env_103123/lib/python3.10/site-packages/scipy/interpolate/_interpolate.py", line 12, in <module>
    from . import _fitpack_py
  File "/home/sophie.hourihane/.conda/envs/bw_cpp_env_103123/lib/python3.10/site-packages/scipy/interpolate/_fitpack_py.py", line 10, in <module>
    from ._bsplines import BSpline
  File "/home/sophie.hourihane/.conda/envs/bw_cpp_env_103123/lib/python3.10/site-packages/scipy/interpolate/_bsplines.py", line 9, in <module>
    from scipy.optimize import minimize_scalar
  File "/home/sophie.hourihane/.conda/envs/bw_cpp_env_103123/lib/python3.10/site-packages/scipy/optimize/__init__.py", line 411, in <module>
    from ._root import *
ModuleNotFoundError: No module named 'scipy.optimize._root'
```

### SciPy/NumPy/Python version and system information

```shell
Python 3.10.13 | packaged by conda-forge | (main, Oct 26 2023, 18:07:37) [GCC 12.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info); scipy.show_config()
1.11.3 1.26.0 sys.version_info(major=3, minor=10, micro=13, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    include directory: /home/sophie.hourihane/.conda/envs/bw_cpp_env_103123/include
    lib directory: /home/sophie.hourihane/.conda/envs/bw_cpp_env_103123/lib
    name: blas
    openblas configuration: unknown
    pc file directory: /home/sophie.hourihane/.conda/envs/bw_cpp_env_103123/lib/pkgconfig
    version: 3.9.0
  lapack:
    detection method: pkgconfig
    found: true
    include directory: /home/sophie.hourihane/.conda/envs/bw_cpp_env_103123/include
    lib directory: /home/sophie.hourihane/.conda/envs/bw_cpp_env_103123/lib
    name: lapack
    openblas configuration: unknown
    pc file directory: /home/sophie.hourihane/.conda/envs/bw_cpp_env_103123/lib/pkgconfig
    version: 3.9.0
  pybind11:
    detection method: pkgconfig
    include directory: /home/sophie.hourihane/.conda/envs/bw_cpp_env_103123/include
    name: pybind11
    version: 2.11.1
Compilers:
  c:
    commands: /home/conda/feedstock_root/build_artifacts/scipy-split_1696467628975/_build_env/bin/x86_64-conda-linux-gnu-cc
    linker: ld.bfd
    name: gcc
    version: 12.3.0
  c++:
    commands: /home/conda/feedstock_root/build_artifacts/scipy-split_1696467628975/_build_env/bin/x86_64-conda-linux-gnu-c++
    linker: ld.bfd
    name: gcc
    version: 12.3.0
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 0.29.36
  fortran:
    commands: /home/conda/feedstock_root/build_artifacts/scipy-split_1696467628975/_build_env/bin/x86_64-conda-linux-gnu-gfortran
    linker: ld.bfd
    name: gcc
    version: 12.3.0
  pythran:
    include directory: ../../_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/python3.10/site-packages/pythran
    version: 0.14.0
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
  cross-compiled: false
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
Python Information:
  path: /home/sophie.hourihane/.conda/envs/bw_cpp_env_103123/bin/python
  version: '3.10'
```
```

**Comment by sophiehourihane** [comment]
Also fixed by running 

``` 
mamba install scipy=1.11.2
```

**Comment by lucascolley** [comment]
I can't reproduce this - is the problem still there after running `mamba install scipy=1.11.3`?

**Comment by rgommers** [comment]
@sophiehourihane we need some more info here to be able to help, in particular how you installed the failing `scipy` package and if that is reproducible now that you've got it working once (I suspect not).

**Comment by sophiehourihane** [comment]
This was the environment I installed from: I believe the issue was occurring because of some interaction between gwpy and scipy. 

channels:
- conda-forge
dependencies:
- c-compiler
- cxx-compiler
# for now, install cppyy through pip because the version from conda is out of date and has bugs
# - cppyy # Must be installed after cxx compiler
- clang
- clang-tools
- catch2
- cmake
- make
- h5py
- llvm-openmp
- ezdag
- eigen
- autodiff
- pkg-config
- fftw * nompi*
- gsl
- gwpy
- liblal
- liblalframe
- liblalinference
- liblalsimulation
- pandas
- pytables
- python
- python-clang
- rapidjson
- matplotlib
- numpy
- scipy
name: bayeswave-cpp

**Comment by rgommers** [comment]
> Also fixed by running `mamba install scipy=1.11.2`
> 
> I believe the issue was occurring because of some interaction between gwpy and scipy.

Thank you for the explanation and confirming that it's fixed now after a reinstall. 

`gwpy` indeed contains a `def signal`, so some kind of mix-up seems possible indeed. There shouldn't be anything left to fix/change here, so I will close this issue.
